### PR TITLE
LPS-68743 Unit test for display context.

### PR DESCRIPTION
**Pull request by brandizzi**

### CODE REVIEW

File: modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-service/src/main/java/com/liferay/dynamic/data/lists/internal/upgrade/v1_0_2/UpgradeDDLRecordSetSettings.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+			createRequireAuthenticationSetting();
+
+		fieldValues.put(requireAuthenticationSetting);
+
+		settingsJSONObject.put("fieldValues", fieldValues);
+
+		return settingsJSONObject.toJSONString();
+	}
+
+	protected boolean containsRequireAuthenticationSetting(
```
Comment (marcellustavares) [inline modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-service/src/main/java/com/liferay/dynamic/data/lists/internal/upgrade/v1_0_2/UpgradeDDLRecordSetSettings.java]: @brandizzi, don't think we need to add this checking. We'll never have this field added in this schema-version.

File: modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-form-web/src/main/java/com/liferay/dynamic/data/lists/form/web/internal/display/context/DDLFormAdminDisplayContext.java [hunk 2]
```diff
@@ -350,6 +351,22 @@ public DDLRecordVersion getRecordVersion() throws PortalException {
...
+	public String getRestrictedFormURL() {
+		StringBundler sb = new StringBundler(4);
+
+		ThemeDisplay themeDisplay =
+			_ddlFormAdminRequestHelper.getThemeDisplay();
+
+		Group group = themeDisplay.getSiteGroup();
+
+		sb.append(themeDisplay.getPortalURL());
+		sb.append(group.getPathFriendlyURL(true, themeDisplay));
```
Comment (marcellustavares) [inline modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-form-web/src/main/java/com/liferay/dynamic/data/lists/form/web/internal/display/context/DDLFormAdminDisplayContext.java]: Had a hard time to spot the difference between these 2 methods. Couldn't we create a third method to hold the duplicated lines?

File: modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/internal/upgrade/v1_0_2/test/UpgradeDDLRecordSetSettingsTest.java [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+			if (fieldName.equals(field)) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	protected JSONArray createFieldValues(boolean hasSetting) {
+		JSONArray array = _jsonFactory.createJSONArray();
```
Comment (marcellustavares) [inline modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/internal/upgrade/v1_0_2/test/UpgradeDDLRecordSetSettingsTest.java]: Please rename from `array` to `jsonArray`

File: modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/internal/upgrade/v1_0_2/test/UpgradeDDLRecordSetSettingsTest.java [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+		recordSet = getRecordSet(recordSet);
+
+		fieldValues = getFieldValues(recordSet.getSettings());
+
+		Assert.assertTrue(containsField(fieldValues, "requireAuthentication"));
+	}
+
+	@Test
+	public void testDoNotRepeatRequireAuthenticationSetting() throws Exception {
```
Comment (marcellustavares) [inline modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/internal/upgrade/v1_0_2/test/UpgradeDDLRecordSetSettingsTest.java]: I tend to think the code and this test is not needed.

File: modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/internal/upgrade/v1_0_2/test/UpgradeDDLRecordSetSettingsTest.java [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+		array.put(languageId);
+
+		return array;
+	}
+
+	protected JSONObject getFieldValue(
+		String instanceId, String name, String value) {
+
+		JSONObject object = _jsonFactory.createJSONObject();
```
Comment (marcellustavares) [inline modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/internal/upgrade/v1_0_2/test/UpgradeDDLRecordSetSettingsTest.java]: Please rename from `object` to `jsonObject`

**Comment by brandizzi** [comment]
Changes applied. Re-sent as https://github.com/marcellustavares/liferay-portal/pull/1784. Thanks!

**Pull request closed without merging**